packages/experiment-tag/src/experiment.ts

```diff
@@ -226,14 +235,24 @@ const handleInject = (action, key: string, variant: Variant) => {
     }
     // Inject
     const utils = getInjectUtils();
-    const js = action.data.js;
     appliedInjections.add(id);
-    inject(js, html, utils, id);
+    try {
+      const fn = globalScope[id];
+      if (fn && typeof fn === 'function') {
+        fn(html, utils, id);
+      }
+    } catch (e) {
+      console.error(
```

Comment:
Do we want to remove the script and style html tag elements from document if inject failed?

Reply:
Yeah we should probably remove the script. Style may still be useful though.